### PR TITLE
Hello world - Front End

### PR DESCRIPTION
**Pull request by TadesseDev**
# Changes Introduced 😎
---
- Create a new React app called 'hello-react-front-end' ✔
- Create the App component with react-router. ✔
- Create the Greeting component that will display a greeting. Set it up as a route in your App component. ✔
- Create a store, an action and a reducer that will connect to My API endpont to get the random greeting. ✔
- Display the random greeting in My Greeting component. ✔

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
-    <App />
-  </React.StrictMode>
+    <BrowserRouter>
+      <Provider store={store}>
+        <Routes>
+          <Route path="/greeting" element={<Greeting />} />
+          <Route path="/" element={<App />} />
+        </Routes>
+      </Provider>
+    </BrowserRouter>
```
Comment (ShahierNashaat) [inline src/index.js]: - [ ] Kindly notice that the `App` component has a route that leads to the `Greeting` component, not the `index` component.

File: src/Redux/reducers/Greeting/greeting.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+const RANDOM_GREET_ACTION = 'Greeting/greeting/RANDOM_GREET_ACTION';
+
+export const generateGreeting = () => (dispatch) => {
+  const myHeaders = new Headers();
+  myHeaders.append('Content-Type', 'application/json');
+  myHeaders.append('Accept', 'application/json');
+  fetch('http://localhost:4000/greeting', {
```
Comment (ShahierNashaat) [inline src/Redux/reducers/Greeting/greeting.js:7]: - [ ] Kindly notice that you should write the API URL right `http://localhost:400/greeting`.

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @Tadesse-Alemayehu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (TadesseDev) [reply src/index.js]: Thank you @ShahierNashaat. Ill make the changes.

File: src/Redux/reducers/Greeting/greeting.js (hunk 2 above)
Comment (TadesseDev) [reply src/Redux/reducers/Greeting/greeting.js:7]: Can you elaborate more on this. @ShahierNashaat. What do you mean by `right`, currently my rails is running on local host with port 4000. How should I rephrase it?

Comment (ShahierNashaat) [reply src/Redux/reducers/Greeting/greeting.js:7]: As you wrote on your Readme file to choose the 400 port but you wrote 4000 in the front end, it doesn't matter but please update the readme file at the back end repo.

Comment (TadesseDev) [reply src/Redux/reducers/Greeting/greeting.js:7]: I see—ill correct that. Thank you @ShahierNashaat

Comment (TadesseDev) [reply src/Redux/reducers/Greeting/greeting.js:7]: Fixed with [3eaa1f7](https://github.com/Tadesse-Alemayehu/Hello-Rails-Back-End/pull/1/commits/3eaa1f78e5a792649883798b5075be2638901b12) commit hash.

File: src/index.js (hunk 1 above)
Comment (TadesseDev) [reply src/index.js]: Fixed on [bab1df5](https://github.com/Tadesse-Alemayehu/Hello-React-Front-End/pull/1/commits/bab1df52ebf1e187e93000bde2a4f0f57028bd24) commit hash.

**Review by ShahierNashaat — APPROVED** [review]
Hi @Tadesse-Alemayehu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Good overall design ✔️
- Your code is clean and well documented ✔️ 

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by TadesseDev** [comment]
Thank you @ShahierNashaat. Nice to have your approval.

**Pull request merged**